source/elements/oneDPL/source/parallel_api/parallel_range_api.rst

```diff
@@ -344,5 +460,79 @@ Mutating Operations
 
   }
 
+In-place Mutating Operations
+++++++++++++++++++++++++++++
+
+.. code:: cpp
+
+  // Defined in <oneapi/dpl/algorithm>
+
+  namespace oneapi::dpl::ranges {
+
+    // fill
+    template <typename ExecutionPolicy, std::ranges::random_access_range R,
+              typename T = std::ranges::range_value_t<R>>
+      requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<ExecutionPolicy>> &&
+               std::ranges::sized_range<R> &&
+               std::indirectly_writable<std::ranges::iterator_t<R>, const T&>
+      std::ranges::borrowed_iterator_t<R>
+        fill (ExecutionPolicy&& pol, R&& r, const T& value);
+  
+    // generate
+    template <typename ExecutionPolicy, std::ranges::random_access_range R, std::copy_constructible F>
+      requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<ExecutionPolicy>> &&
+               std::ranges::sized_range<R> && std::invocable<F&> &&
+               std::indirectly_writable<std::ranges::iterator_t<R>, std::invoke_result_t<F&>>
+      std::ranges::borrowed_iterator_t<R>
+        generate (ExecutionPolicy&& pol, R&& r, F gen);
+
+    // replace
+    template <typename ExecutionPolicy, std::ranges::random_access_range R,
+              typename Proj = std::identity,
+              typename T1 = /*projected-value-type*/<std::ranges::iterator_t<R>, Proj>, typename T2 = T1>
```

Comment:
For my own understanding...
What is the point of defaulted template arguments here without default function arguments (wont they always be overridden by the deduced type)?

Are there situations where this may resolve some ambiguity when attempting to deduce the types `T1`, `T2` from the function arguments?

Is it merely to describe the "expected type"?

Reply:
Ah, nevermind. I found a couple answers.  One would be in the context of a function pointer, another would be providing arguments with something like this `{}` which can only be deduced with a default template arg.